packages/headless/src/controllers/facets/facet-search/specific/headless-facet-search.ts

```diff
@@ -45,6 +46,11 @@ export function buildFacetSearch(
       genericFacetSearch.select(value);
     },
 
+    singleSelect(value: SpecificFacetSearchResult) {
+      dispatch(singleSelectFacetSearchResult({facetId, value}));
```

Comment:
Going back to [this thread](https://github.com/coveo/ui-kit/pull/874#discussion_r648559754), I believe adjusting the specific facet `deselectAll` slice handler to use that of the range facets would make it possible single-select by composing a deselectAll with a toggleSelect/toggleFacetSearchResult.

Composing would be much less source and unit test code.
If we want to one day support custom ordering on the specific facet, we would need to adjust the deselectAll handler anyway.

If there is a roadblock with the approach, I want to understand what it is exactly,

Reply:
Oh i haven’t seen the latest message of that thread.

For the deselect you’re right, we could just set them all to idle, I’ve double checked and the API will refill values there. We can adjust this, it will remove code for sure! I’ll go ahead with it

For custom ordering, in JSUI, it was a simple manual sort of the response, not a ordering during the request. It was deemed sufficient back then https://github.com/coveo/search-ui/pull/1689#discussion_r566160826 If we do the same, don’t think changing deselect all affects this.